### PR TITLE
Exchange auth code for access token.

### PR DESCRIPTION
**Pull request by marcferna**
@drn || @davywentworth

### CODE REVIEW

File: lib/omniauth/strategies/google-oauth2-access-token.rb [hunk 1]
```diff
@@ -81,6 +81,10 @@ def request_phase
       end
 
       def callback_phase
+        if !request.params['auth_code'].to_s.empty? && request.params['access_token'].to_s.empty?
```
Comment (davywentworth) [inline lib/omniauth/strategies/google-oauth2-access-token.rb:84]: that first half would be more clear if it wasn't a dobule negative:
`request.params['auth_code'].to_s.present?`

File: lib/omniauth/strategies/google-oauth2-access-token.rb [hunk 2]
```diff
@@ -163,6 +167,20 @@ def verify_token(id_token, access_token)
...
+          'https://www.googleapis.com/oauth2/v4/token',
+          params: {
+            code:          auth_code,
+            client_id:     ENV['GOOGLE_CLIENT_ID'],
+            client_secret: ENV['GOOGLE_CLIENT_SECRET'],
+            grant_type:    'authorization_code'
+          }
+        ).parsed
+        raw_response['access_token']
+      end
```
Comment (davywentworth) [inline lib/omniauth/strategies/google-oauth2-access-token.rb:182]: if any of this were to except, is that properly handled somewhere else?

**Comment by davywentworth** [comment]
LGTM.

File: lib/omniauth/strategies/google-oauth2-access-token.rb (hunk 1 above)
Comment (marcferna) [inline lib/omniauth/strategies/google-oauth2-access-token.rb:84]: didn't want to use any `rails` helper - this is pure ruby

**Comment by drn** [comment]
huh, yeah - lgtm if it works. this is the standard way of doing this?

**Pull request merged**